gateleen-delegate/src/main/java/org/swisspush/gateleen/delegate/Delegate.java

```diff
@@ -0,0 +1,208 @@
+     * matching the given delegate pattern. Also the
+     * request uri is adapted.
+     *
+     * @param uri original request
+     * @param requestObject the delegate request object
+     * @param doneHandler the done handler called as soon as the request is executed
+     */
+    private void createRequest(final String uri, final JsonObject requestObject, final Handler<HttpClientResponse> doneHandler) {
+        // matcher to replace wildcards with matching groups
+        final Matcher matcher = pattern.matcher(uri);
+
+        // adapt the request uri if necessary
+        final String requestUri = matcher.replaceAll(requestObject.getString(URI));
+
+        // get the string represantion of the payload object
+        String payloadStr;
+        try {
+            payloadStr = requestObject.getString(PAYLOAD);
+        } catch(ClassCastException e) {
+            payloadStr = requestObject.getJsonObject(PAYLOAD).encode();
+        }
+
+        // replacement of matching groups
+        if(payloadStr != null) {
+            payloadStr = matcher.replaceAll(payloadStr);
+        }
+
+        // headers of the delegate
+        MultiMap headers = new CaseInsensitiveHeaders();
+        JsonArray headersArray = requestObject.getJsonArray(HEADERS);
+        if ( headersArray != null ) {
+            if (LOG.isTraceEnabled()) {
+                LOG.trace("Request headers:");
+            }
+
+            headersArray.forEach(header -> {
+                if (LOG.isTraceEnabled()) {
+                    LOG.trace(" > Key [{}], Value [{}]", ((JsonArray) header).getString(0), ((JsonArray) header).getString(1) );
+                }
+
+                headers.add(((JsonArray) header).getString(0),((JsonArray) header).getString(1));
+
+            });
+        }
+
+        HttpClientRequest delegateRequest = selfClient.request(HttpMethod.valueOf(requestObject.getString(METHOD)), requestUri, doneHandler);
+        delegateRequest.headers().setAll(headers);
+        delegateRequest.exceptionHandler(exception -> LOG.warn("Delegate request {} failed: {}",requestUri , exception.getMessage()));
+        delegateRequest.setTimeout(120000); // avoids blocking other requests
+        delegateRequest.end(Buffer.buffer(payloadStr));
+    }
+
+    /**
+     * Creates a new DoneHandler for the given request.
+     *
+     * @param request the original request
+     * @return a doneHandler
+     */
+    private Handler<HttpClientResponse> installDoneHandler(final HttpServerRequest request) {
+        return new Handler<HttpClientResponse>() {
+            private AtomicInteger currentIndex = new AtomicInteger(0);
+
+            @Override
+            public void handle(HttpClientResponse response) {
+                if ( LOG.isTraceEnabled() ) {
+                   LOG.trace("Done handler - handle");
+                }
+
+                // request was fine
+                if ( response.statusCode() == StatusCode.OK.getStatusCode() ||
```

Comment:
Here we should accept all 2xx response codes, not only 200 and 202. We may have a 204, for example.